packages/react/src/template.tsx

```diff
@@ -27,7 +33,7 @@ export type ReferencedDataState = {
   ref: Path
 }
 type EventDataState = ActionEventDataState | HandlerEventDataState
-export type ActionEventDataState<T = any> = {
+export type ActionEventDataState<T = string | string[]> = {
```

Comment:
Yeah, should be opaque. This type annotation just changes the default value. However, since it's opaque, we shouldn't probably make assumptions about its type! Will double check this.

Reply:
Oh yeah hmm, probably not a big deal. For some reason I misread this as `extends`